packages/firestore/src/api/field_value.ts

```diff
@@ -123,7 +123,7 @@ export class ServerTimestampFieldValueImpl extends SerializableFieldValue {
   }
 
   _toFieldTransform(context: ParseContext): FieldTransform {
-    return new FieldTransform(context.path!, ServerTimestampTransform.instance);
```

Comment:
Removed static instance member as these are not tree-shakeable. See https://github.com/firebase/firebase-js-sdk/pull/3264

Reply:
I noticed we reference all the transforms inside of `transform_operation.ts`. Will the branches in that file get erased if we don't reference `ServerTimestampTransform`? It seems like we have a problem either way. Am I missing something?

Reply:
Given the way our serializer works (we look at the data in IndexedDB and then decide which transform to create), I don't think it is possible to structure our SDK in a way that `ServerTimestampTransform` is not included if another transform is used. Having `ServerTimestampTransform.instance` however means that ServerTimestampTransform may also be included in the final bundle if neither the serializer not the transforms are used at all.

Reply:
That's what I figured. imho - I think that if the final bundle takes the hit of including these types regardless of use we should consider maintaining the existing pattern. I was never a fan of the new style and and if we're not getting any benefit from it, why deviate?

Reply:
The Lite SDK doesn't reference these methods and is able to remove them (see the binary size report: -7%). Does that answer your concern?